extensions/README.md

```diff
@@ -0,0 +1,362 @@
+[CoRIM
+specification](https://www.ietf.org/archive/id/draft-ietf-rats-corim-02.htm)
+may be extended by other specifications at well defined points identified in the
+spec using CDDL extension sockets. This implementation, likewise, allows
+dependent code to register extension types. This is done via three distinct
+extension mechanisms:
+
+1. Some structures embed an Extensions object that allows registering a
+   user-provided struct. The fields of that struct will be treated as extension
+   fields to embedding structure. This corresponds to the map-extension sockets
+   in the spec.
+2. Some elements can contain values of one of pre-determined types. New types
+   can be provided by registering a factory function for those types using an
+   appropriate function. This (mostly) corresponds to the type-choice extension
+   sockets in the spec.
+3. A couple of type-choice sockets (`$tag-rel-type-choice`,
+   `$corim-role-type-choice` and `$comid-role-type-choice`) define what, in
+   effect, are extensible enums. They allow providing additional values, rather
+   than types. This implementation provides registration functions for new
+   values for those types.
+
+> [!NOTE]
+> The CoRIM spec includes some CDDL from the CoSWID spec that also features
+> extension sockets. This code base does include the CoSWID implementation, and
+> so does not cover those extension points.
+
+
+## Map Extensions
```

Comment:
So Far we have been using the term `structures` for extending CoRIM Maps

Reply:
Theere are no structures in CoRIM, in the problem domain/concemptually, they are CoRIM map extensions, that are implemented suing `Extensions` `structs`. "Map" and "type choice" are the names of the extensions types being implemented.